### DISCUSSION TITLE
Taiko/Gauge - The Header Basic Authorization is getting mixed with the form Login

### DISCUSSION THREAD

**Discussion by jawaharsan** [getgauge/taiko#2263] [Development]
Hi,

We try to implement Basic Authorization with Base64 encoding method to launch a website. It works fine. But, when we try to login as a user which is a form on the application flow, it is not getting logged in. When we do it manually, there is no issue. Kindly help.

The code is something like this.

goto(url,{headers:{'Authorization':'Basic <base64encodedcredsvalue>'}}) => This works well.

On the login page, I need to give the user name (EmailID) and password as below:
await write("EmailID", into(textBox{id:"IDValueEmail"}));
await write("Password", into(textBox{id:"IDValuePassword"}));
await evaluate($(customelocatorSignin), ele => ele.click());

I have two questions:
1. How to overwrite the password set for the URL using headers?
2. I am not able to login with the particular valid credentials as a user on the login page. It just stays on the same page without throwing any error. How to resolve this issue.

Any pointer to the above questions would highly be appreciated. Thank you.

Regards,
Santhosh.

**Comment by sriv** [comment]
The HTTP basic auth requires the `Authorization` header. you can set it explicitly when doing a `goto` but if you need to set it when you click the login button, then you will have to intercept the request and add the auth headers.

See https://docs.taiko.dev/api/intercept/
